src/device/simulator.ts

```diff
@@ -286,11 +308,19 @@ export class SimulatorDeviceConnection
   }
 
   radioSend(message: string) {
+    const kind = "radio_input";
     const data = new TextEncoder().encode(message);
     const prefixed = new Uint8Array(3 + data.length);
     prefixed.set([1, 0, 1]);
     prefixed.set(data, 3);
-    this.postMessage("radio_input", { data: prefixed });
+    this.postMessage(kind, { data: prefixed });
+    if (!this.sensorsLogged[kind]) {
+      this.logging.event({
+        type: "sim-user",
+        message: kind,
+      });
+      this.logSensor(kind);
+    }
```

Comment:
Maybe dedup to a (private) logSensor call on `this`?

Reply:
[`2dc5220`](https://github.com/microbit-foundation/python-editor-v3/pull/1021/commits/2dc52203f09a15db813d0952316697974d8ce5d4).